src/Network/Response.php

```diff
@@ -1348,23 +1348,48 @@ public function cookie($options = null)
      * @param string|array $allowedDomains List of allowed domains, see method description for more details
      * @param string|array $allowedMethods List of HTTP verbs allowed
      * @param string|array $allowedHeaders List of HTTP headers allowed
+     * @param array $options Params for specific headers
+     *
      * @return void
      */
-    public function cors(Request $request, $allowedDomains, $allowedMethods = [], $allowedHeaders = [])
-    {
+    public function cors(Request $request, $allowedDomains, $allowedMethods = [], $allowedHeaders = [], array $options = []) {
         $origin = $request->header('Origin');
         if (!$origin) {
             return;
         }
 
-        $allowedDomains = $this->_normalizeCorsDomains((array)$allowedDomains, $request->is('ssl'));
+        // For BC. This can be removed in 4.x when the $allowedDomains, $allowedMethods, $allowedHeaders params are gone.
+        $options += [
+            'allowedDomains' => (array)$allowedDomains,
+            'allowedMethods' => (array)$allowedMethods,
+            'allowedHeaders' => (array)$allowedHeaders
+        ];
```

Comment:
Is it useful to use func_num_args() and accept option in the 2nd argument? Another option could be to add methods for each cors related feature?